### PR TITLE
Add demonstration of file configuration to the javaagent example

### PR DESCRIPTION
**Pull request by jack-berg**

### CODE REVIEW

File: javaagent/docker-compose.yml [hunk 1]
```diff
@@ -4,19 +4,24 @@ services:
     build: ./
     environment:
       OTEL_SERVICE_NAME: "agent-example-app"
-      OTEL_EXPORTER_OTLP_ENDPOINT: "http://collector:4317"
+      OTEL_EXPORTER_OTLP_ENDPOINT: "http://collector:4318"
```
Comment (breedx-splk) [inline javaagent/docker-compose.yml:7]: This is the default for 2.0 right....so do we like it to be explicit or can we just omit it?

File: javaagent/sdk-config.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+resource:
+  attributes:
+    service.name: agent-example-app
+
+logger_provider:
+  processors:
+    - batch:
+        exporter:
+          otlp:
+            endpoint: http://collector:4317
```
Comment (breedx-splk) [inline javaagent/sdk-config.yaml:12]:
```suggestion
            endpoint: http://collector:4318
```

File: javaagent/sdk-config.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+tracer_provider:
+  processors:
+    - batch:
+        exporter:
+          otlp:
+            endpoint: http://collector:4317
```
Comment (breedx-splk) [inline javaagent/sdk-config.yaml:19]:
```suggestion
            endpoint: http://collector:4318
```

File: javaagent/sdk-config.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+meter_provider:
+  readers:
+    - periodic:
+        exporter:
+          otlp:
+            endpoint: http://collector:4317
```
Comment (breedx-splk) [inline javaagent/sdk-config.yaml:26]:
```suggestion
            endpoint: http://collector:4318
```

**Review by breedx-splk — APPROVED** [review]
Nice. I find having concrete simple examples like this really helpful. Just get those ports corrected I guess and otherwise it LGMT. Thanks.

File: javaagent/docker-compose.yml (hunk 1 above)
Comment (jack-berg) [reply javaagent/docker-compose.yml:7]: The endpoint defaults to `https://localhost:4318`, we need `http://collector:4318` 🙂

File: javaagent/sdk-config.yaml (hunk 2 above)
Comment (jack-berg) [reply javaagent/sdk-config.yaml:12]: Need to update the protocol to http/protobuf along with this. I've gone ahead and done that.

**Pull request merged**